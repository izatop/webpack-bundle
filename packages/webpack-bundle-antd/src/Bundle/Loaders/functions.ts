import tsImportPluginFactory from "ts-import-plugin";

export function getCustomTransformers() {
    return {
        before: [
            tsImportPluginFactory({
                libraryDirectory: "es",
                libraryName: "antd",
                style: "css",
            }),
        ],
    };
}
