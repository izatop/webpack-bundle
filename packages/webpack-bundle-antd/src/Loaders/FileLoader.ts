import {Loader} from "webpack-bundle";

export const FILE_LOADER_EXTENSION_TEST = /\.(woff(2)?|ttf|eot|svg|jpg|jpeg|png|gif)(\?v=\d+\.\d+\.\d+)?$/;

export class FileLoader extends Loader {
    constructor(test = FILE_LOADER_EXTENSION_TEST) {
        super({
            test,
            loader: "file",
            options: {
                name: "[hash].[ext]",
                outputPath: "assets/",
            },
        });
    }
}
