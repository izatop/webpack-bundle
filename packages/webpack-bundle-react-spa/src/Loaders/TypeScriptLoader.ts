import * as webpack from "webpack";
import {Loader} from "webpack-bundle";

export class TypeScriptLoader extends Loader {
    constructor(options: webpack.RuleSetQuery = {}) {
        super({
            options,
            loader: "ts",
            test: /\.tsx?/,
        });
    }
}
