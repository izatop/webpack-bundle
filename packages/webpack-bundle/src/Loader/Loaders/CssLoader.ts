import {Loader} from "../Loader";

export class CssLoader extends Loader {
    protected getDefaults() {
        return {
            test: /\.css$/,
            use: "css",
        };
    }
}
