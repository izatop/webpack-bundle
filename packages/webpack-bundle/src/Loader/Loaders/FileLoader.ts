import {Loader} from "../Loader";

export class FileLoader extends Loader {
    protected getDefaults() {
        return {
            test: /\.(woff(2)?|ttf|eot|svg|jpg|jpeg|png|gif)(\?v=\d+\.\d+\.\d+)?$/,
            loader: "file-loader",
            type: "asset/resource",
            options: {
                name: "[hash].[ext]",
                outputPath: "assets/",
            },
        };
    }
}
