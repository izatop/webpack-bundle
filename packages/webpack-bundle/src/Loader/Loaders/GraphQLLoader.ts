import {Loader} from "../Loader";

export class GraphQLLoader extends Loader {
    protected getDefaults() {
        return {
            test: /\.(graphql|gql)$/,
            exclude: /node_modules/,
            loader: 'graphql-tag/loader'
        };
    }
}
