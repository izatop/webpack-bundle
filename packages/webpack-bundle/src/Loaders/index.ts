export * from "./Loader";
export * from "./TypeScript/TypeScriptLoader";
