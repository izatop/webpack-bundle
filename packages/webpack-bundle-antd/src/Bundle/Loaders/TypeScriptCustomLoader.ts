import {IWebpackValue, Loaders} from "webpack-bundle";
import {getCustomTransformers} from "./functions";

export class TypeScriptCustomLoader extends Loaders.TypeScriptLoader {
    constructor(mode: IWebpackValue<"mode">) {
        super({
            options: {
                transpileOnly: !(mode === "production"),
                getCustomTransformers,
            },
        });
    }
}
