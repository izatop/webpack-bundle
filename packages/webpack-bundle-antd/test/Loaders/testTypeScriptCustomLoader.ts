import {getCustomTransformers} from "../../src/Bundle/Loaders/functions";
import {TypeScriptCustomLoader} from "../../src/Bundle/Loaders/TypeScriptCustomLoader";

it("Match TypeScriptCustomLoader options", () => {
    const loader = new TypeScriptCustomLoader("development");
    expect(loader.getValue()!.options)
        .toMatchObject({getCustomTransformers});

    expect(getCustomTransformers())
        .toMatchSnapshot();
});
