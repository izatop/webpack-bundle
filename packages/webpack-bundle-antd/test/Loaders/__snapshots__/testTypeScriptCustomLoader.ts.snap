// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match TypeScriptCustomLoader options 1`] = `
Object {
  "before": Array [
    [Function],
  ],
}
`;
